libraries/lib-ipc/IPCServer.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <memory>
+
+class IPCChannel;
+class IPCChannelStatusCallback;
+
+/**
+ * \brief Simple TCP socket based ipc server.
+ * Currently, implementation allows only one channel per
+ * instance. IPCChannelStatusCallback methods are always called
+ * from the worker thread, though IPCChannel instance created
+ * by this implementation is thread safe (considering lifetime limitations)
```

Comment:
The IPCServer object itself is not thread safe -- I mean `CreateChannel` could race to assign `Impl::mWorker` if two threads called it.  Maybe that is worth a comment.